### PR TITLE
Dump exceptions live when they happen when testing tests with flutter run

### LINKED ISSUE #7669: LiveTestWidgetsFlutterBinding should print a message when setting _pendingExceptionDetails
It would save a lot of time when debugging tests

### PR DESCRIPTION
**Pull request by Hixie**
Fixes https://github.com/flutter/flutter/issues/7669

### CODE REVIEW

**Comment by Hixie** [comment]
cc @yjbanov for review

File: packages/flutter_test/lib/src/binding.dart [hunk 1]
```diff
@@ -375,6 +387,7 @@ abstract class TestWidgetsFlutterBinding extends BindingBase
         );
       } else {
         _pendingExceptionDetails = details;
+        reportExceptionNoticed(details); // mostly this is just a hook for the LiveTestWidgetsFlutterBinding
```
Comment (yjbanov) [inline packages/flutter_test/lib/src/binding.dart]: Not sure if it matters, but the docs say that `reportExceptionNoticed ` is called "when there is no pending exception". So perhaps this line should run before we set `_pendingExceptionDetails`?

**Review by yjbanov — APPROVED** [review]

**Comment by yjbanov** [comment]
lgtm

File: packages/flutter_test/lib/src/binding.dart (hunk 1 above)
Comment (Hixie) [reply packages/flutter_test/lib/src/binding.dart]: Yeah, that's reasonable. Will switch that around.

**Pull request merged**